examples/allennlp/allennlp_simple.py

```diff
@@ -97,29 +104,38 @@ def objective(trial):
     if DEVICE > -1:
         model.to(torch.device("cuda:{}".format(DEVICE)))
 
-    lr = trial.suggest_loguniform("lr", 1e-1, 1e0)
+    lr = trial.suggest_loguniform("lr", 1e-2, 1e-1)
     optimizer = torch.optim.SGD(model.parameters(), lr=lr)
 
-    iterator = allennlp.data.iterators.BasicIterator(batch_size=10,)
-    iterator.index_with(vocab)
+    data_loader = torch.utils.data.DataLoader(
+        train_dataset, batch_size=10, collate_fn=allennlp.data.allennlp_collate
+    )
+    validation_data_loader = torch.utils.data.DataLoader(
+        valid_dataset, batch_size=10, collate_fn=allennlp.data.allennlp_collate
+    )
 
     serialization_dir = os.path.join(MODEL_DIR, "trial_{}".format(trial.number))
-    trainer = allennlp.training.Trainer(
+    trainer = allennlp.training.GradientDescentTrainer(
         model=model,
         optimizer=optimizer,
-        iterator=iterator,
-        train_dataset=train_dataset,
-        validation_dataset=valid_dataset,
-        patience=3,
-        num_epochs=6,
+        data_loader=data_loader,
+        validation_data_loader=validation_data_loader,
+        validation_metric="+accuracy",
+        patience=None,
```

Comment:
[question] Why `3` to `None`?

Reply:
`GradientDescentTrainer` works with early stopping by default.
I update a value of patience, as early stopping could conflict with `AllenNLPPruningCallback`.
(Thank you for the question, I put a comment since it should be mentioned in a script)

Reply:
Much appreciate your clear explanation.